src/modules/aws_ecs_fargate@0.0.1/index.ts

```diff
@@ -131,10 +131,13 @@ export const AwsEcsFargateModule: Module = new Module({
       if (s.networkConfiguration?.awsvpcConfiguration) {
         const networkConf = s.networkConfiguration.awsvpcConfiguration;
         out.assignPublicIp = networkConf.assignPublicIp;
-        out.securityGroups = networkConf.securityGroups?.length ?
-          await AwsSecurityGroupModule.mappers.securityGroup.db.read(ctx, networkConf.securityGroups) ??
-            await AwsSecurityGroupModule.mappers.securityGroup.cloud.read(ctx, networkConf.securityGroups)
-          : []
+        const securityGroups = [];
+        for (const sg of networkConf.securityGroups ?? []) {
+          securityGroups.push(await AwsSecurityGroupModule.mappers.securityGroup.db.read(ctx, sg) ??
+            await AwsSecurityGroupModule.mappers.securityGroup.cloud.read(ctx, sg));
+        }
+        if (securityGroups.filter(sg => !!sg).length !== networkConf.securityGroups.length) throw new Error('Security groups need to be loaded first')
```

Comment:
Yes, if it fails it goes back expecting to find it the next iteration, if the same error happens again and there's nothing else to do it will `throw`.